src/dfx/src/lib/builders/mod.rs

```diff
@@ -351,21 +353,25 @@ pub fn run_command(args: Vec<String>, vars: &[Env<'_>], cwd: &Path) -> DfxResult
 
 /// Set the permission of the given file to be writeable.
 pub fn set_perms_readwrite(file_path: &PathBuf) -> DfxResult<()> {
-    let mut perms = std::fs::metadata(file_path)
-        .with_context(|| {
+    #[cfg(not(target_os = "windows"))]
```

Comment:
I don't understand the documentation. Other places in the repo only care about permissions on unix, so this could be the way to do it?

Reply:
The intent of this method is to make a file owner-writable.  See https://github.com/dfinity/sdk/pull/3017 for where this was necessary in order to avoid `Permission denied` errors in post-build.

One day we may have a functioning windows build, but until then, it's fine if this method only does anything on cfg(unix).